server/src/main/java/org/elasticsearch/index/get/GetResult.java

```diff
@@ -222,20 +226,34 @@ public String sourceAsString() {
         return sourceAsMap();
     }
 
+
+    public Map<String, DocumentField> getMetadataFields() {
+        return new HashMap<>(metaFields);
+    }
+
+    public Map<String, DocumentField> getDocumentFields() {
+        return new HashMap<>(documentFields);
+    }
+
```

Comment:
I am not very happy to introduce these getters, but they are required in `UpdateResponse`.  As of now, we have 3 different getters - 2 new for meta/nonmeta and 1 old for all aggregate fields